src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -141,30 +140,45 @@ export const SpawnHostModal: React.FC<SpawnHostModalProps> = ({
     }
   }, [awsRegions, publicKeys, dispatch]);
 
-  // Need to initialize these here so they can be used in the useEffect hook
-  let virtualWorkstationDistros = [];
-  let notVirtualWorkstationDistros = [];
+  // recalculate isVirtualWorkstation whenever distro changes
+  // initial distroId can be changed from URL
+  useEffect(() => {
+    dispatch({
+      type: "editDistroEffect",
+      isVirtualWorkstation: !!distrosData?.distros.find(
+        (vd) => distroId === vd.name
+      )?.isVirtualWorkStation,
+    });
+  }, [distroId, dispatch, distrosData?.distros]);
 
   if (distroLoading || publicKeyLoading || awsLoading || volumesLoading) {
     return null;
   }
 
-  const distros = fetchedDistros.filter((d) => d.name.includes(distroInput));
-  virtualWorkstationDistros = distros
-    .filter((d) => d.isVirtualWorkStation)
-    .sort((a, b) => a.name.localeCompare(b.name));
-  notVirtualWorkstationDistros = distros
-    .filter((d) => !d.isVirtualWorkStation)
+  const allDistroOptionsMinusCurrentSelection = (distrosData?.distros ?? [])
+    .filter((d) => d.name.includes(distroInput))
     .sort((a, b) => a.name.localeCompare(b.name));
 
+  const virtualWorkstationDistrosMinusCurrentSelection = allDistroOptionsMinusCurrentSelection.filter(
+    (d) => d.isVirtualWorkStation
+  );
+
+  const notVirtualWorkstationDistrosMinusCurrentSelection = allDistroOptionsMinusCurrentSelection.filter(
+    (d) => !d.isVirtualWorkStation
+  );
```

Comment:
what I have actually makes the most sense for me otherwise it is easy to miss that the lists derived from `fetchedDistros.filter((d) => d.name.includes(distroInput))` excludes the current selection

Reply:
@khelif96 I realized I was wrong about the `excludes the current selection` bit. I'll change the name to "filteredDistros"